src/liballoc/range.rs

```diff
@@ -63,9 +63,74 @@ pub trait RangeArgument<T: ?Sized> {
     /// # }
     /// ```
     fn end(&self) -> Bound<&T>;
-}
 
-// FIXME add inclusive ranges to RangeArgument
+    /// Returns `true` if `item` is contained in the range.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(range_contains)]
+    /// #![feature(collections_range)]
+    ///
+    /// use std::collections::range::RangeArgument;
+    ///
+    /// assert!(!(3..5).contains(&2));
+    /// assert!( (3..5).contains(&3));
+    /// assert!( (3..5).contains(&4));
+    /// assert!(!(3..5).contains(&5));
+    ///
+    /// assert!(!(3..3).contains(&3));
+    /// assert!(!(3..2).contains(&3));
+    ///
+    /// assert!(!(3..).contains(&2));
+    /// assert!( (3..).contains(&3));
+    /// assert!( (3..).contains(&1_000_000_000));
+    ///
+    /// assert!( (..5).contains(&-1_000_000_000));
+    /// assert!( (..5).contains(&4));
+    /// assert!(!(..5).contains(&5));
+    ///
+    /// assert!(!(3..=5).contains(&2));
+    /// assert!( (3..=5).contains(&3));
+    /// assert!( (3..=5).contains(&4));
+    /// assert!( (3..=5).contains(&5));
+    /// assert!(!(3..=5).contains(&6));
+    ///
+    /// assert!( (3..=3).contains(&3));
+    /// assert!(!(3..=2).contains(&3));
+    ///
+    /// assert!( (..=5).contains(&-1_000_000_000));
+    /// assert!( (..=5).contains(&5));
+    /// assert!(!(..=5).contains(&6));
+    /// ```
```

Comment:
Because this is bound only to PartialOrd, I suggest you include some NAN examples in here too.  For example, I would expect `!(0.0..1.0).contains(&f32::NAN)`, `!(0.0..f32::NAN).contains(&0.5)`, and `!(f32::NAN..1.0).contains(&0.5)`.  (And I think that will mean changing the implementation, such as `Included(ref start) => if !(item >= *start) {`, which is the same as what you already have in the totality case, but different in the partial one.)